sql/analyzer/resolve_functions.go

```diff
@@ -17,9 +17,57 @@ package analyzer
 import (
 	"github.com/dolthub/go-mysql-server/sql"
 	"github.com/dolthub/go-mysql-server/sql/expression"
+	"github.com/dolthub/go-mysql-server/sql/grant_tables"
 	"github.com/dolthub/go-mysql-server/sql/plan"
 )
 
+func resolveTableFunctions(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope) (sql.Node, error) {
+	span, _ := ctx.Span("resolve_table_functions")
+	defer span.Finish()
+
+	return plan.TransformUp(n, func(n sql.Node) (sql.Node, error) {
+		if n.Resolved() {
+			return n, nil
+		}
+
+		utf, ok := n.(*expression.UnresolvedTableFunction)
+		if !ok {
+			return n, nil
+		}
+
+		tableFunction, err := a.Catalog.TableFunction(ctx, utf.TableFunctionName())
+		if err != nil {
+			return nil, err
+		}
+
+		database, err := a.Catalog.Database(ctx, ctx.GetCurrentDatabase())
+		if err != nil {
+			return nil, err
+		}
+
+		if privilegedDatabase, ok := database.(grant_tables.PrivilegedDatabase); ok {
+			database = privilegedDatabase.Unwrap()
+		}
+
+		_, err = tableFunction.WithDatabase(database)
+		if err != nil {
+			return nil, err
+		}
+
+		_, err = tableFunction.WithChildren(utf.Children()...)
+		if err != nil {
+			return nil, err
+		}
+
+		_, err = tableFunction.WithExpressions(utf.Arguments...)
```

Comment:
Rather than reuse Expressioner here, I would make this an explicit constructor:

`tbf, err := tableFunction.NewInstance(utf.Arguments)`

WithExpressions has different semantics around e.g. the length of the input arguments and when it gets called.

Reply:
Ahh... seems like there are potential concurrency issues without creating a new instance as well.